domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLogger.kt

```diff
@@ -0,0 +1,142 @@
+  private val enableFastLanguageSwitchingInLesson: PlatformParameterValue<Boolean>,
+  @EnableLoggingLearnerStudyIds
+  private val enableLoggingLearnerStudyIds: PlatformParameterValue<Boolean>,
+  @EnableEditAccountsOptionsUi
+  private val enableEditAccountsOptionsUi: PlatformParameterValue<Boolean>,
+  @EnablePerformanceMetricsCollection
+  private val enablePerformanceMetricsCollection: PlatformParameterValue<Boolean>,
+  @EnableSpotlightUi
+  private val enableSpotlightUi: PlatformParameterValue<Boolean>,
+  @EnableInteractionConfigChangeStateRetention
+  private val enableInteractionConfigChangeStateRetention: PlatformParameterValue<Boolean>,
+  @EnableAppAndOsDeprecation
+  private val enableAppAndOsDeprecation: PlatformParameterValue<Boolean>,
+  @EnableNpsSurvey
+  private val enableNpsSurvey: PlatformParameterValue<Boolean>,
+  @EnableOnboardingFlowV2
+  private val enableOnboardingFlowV2: PlatformParameterValue<Boolean>
+) {
+  /**
+   * A variable containing a list of all the feature flags in the app.
+   *
+   * @return a list of key-value pairs of [String] and [PlatformParameterValue]
+   */
+  private var featureFlagItemMap: Map<String, PlatformParameterValue<Boolean>> = mapOf(
+    DOWNLOADS_SUPPORT to enableDownloadsSupport,
+    EXTRA_TOPIC_TABS_UI to enableExtraTopicTabsUi,
+    LEARNER_STUDY_ANALYTICS to enableLearnerStudyAnalytics,
+    FAST_LANGUAGE_SWITCHING_IN_LESSON to enableFastLanguageSwitchingInLesson,
+    LOGGING_LEARNER_STUDY_IDS to enableLoggingLearnerStudyIds,
+    EDIT_ACCOUNTS_OPTIONS_UI to enableEditAccountsOptionsUi,
+    ENABLE_PERFORMANCE_METRICS_COLLECTION to enablePerformanceMetricsCollection,
+    SPOTLIGHT_UI to enableSpotlightUi,
+    INTERACTION_CONFIG_CHANGE_STATE_RETENTION to enableInteractionConfigChangeStateRetention,
+    APP_AND_OS_DEPRECATION to enableAppAndOsDeprecation,
+    ENABLE_NPS_SURVEY to enableNpsSurvey,
+    ENABLE_ONBOARDING_FLOW_V2 to enableOnboardingFlowV2
+  )
+
+  /**
+   * This method can be used to override the featureFlagItemMap and sets its value to the given map.
+   *
+   * @param featureFlagItemMap denotes the map of feature flag names to their corresponding
+   * [PlatformParameterValue]s
+   */
+  fun setFeatureFlagItemMap(featureFlagItemMap: Map<String, PlatformParameterValue<Boolean>>) {
+    this.featureFlagItemMap = featureFlagItemMap
+  }
+
+  /**
+   * This method logs the name, enabled status and sync status of all feature flags to Firebase.
+   *
+   * @param appSessionId denotes the id of the current appInForeground session
+   */
+  fun logAllFeatureFlags(appSessionId: String) {
+    val featureFlagItemList = mutableListOf<FeatureFlagItemContext>()
+    for (flag in featureFlagItemMap) {
+      featureFlagItemList.add(
+        createFeatureFlagItemContext(flag)
+      )
+    }
+
+    // TODO(#5341): Set the UUID value for this context
+    val featureFlagContext = FeatureFlagListContext.newBuilder()
+      .setAppSessionId(appSessionId)
```

Comment:
Please make sure that the session ID is being verified in tests (it should be stable to check if you set the application seed ID to a fix value like we do in [LoggingIdentifierControllerTest](https://github.com/oppia/oppia-android/blob/14a43d15caa340a13641b57e264248d907cbd98e/domain/src/test/java/org/oppia/android/domain/oppialogger/LoggingIdentifierControllerTest.kt#L328)).

Reply:
I have added tests for the appSessionId in the LoggingIdentifierControllerTest. I assume that's what you meant.